core/common/src/main/java/alluxio/conf/PropertyKey.java

```diff
@@ -2588,6 +2590,7 @@ public String toString() {
           .build();
   public static final PropertyKey LOCALITY_TIER_NODE =
       new Builder(Template.LOCALITY_TIER, Constants.LOCALITY_NODE)
+          .setAlias("alluxio.user.hostname")
```

Comment:
In some previous PR, I found `alluxio.user.hostname` was totally removed and replaced by `alluxio.locality.node`. I actually don't think it is correct. IMO,  `alluxio.locality.node` is not specific to `alluxio.user.hostname` but also applies to `alluxio.{master,worker}.hostname`, right? I think the current implementation is weird by treating `alluxio.{master,worker}.hostname` in parallel with `alluxio.locality.node`

@calvinjia  please let me know what you think. I think we ought to fix this

Reply:
`alluxio.user.hostname` was removed because it was marked as deprecated.

Reply:
right. I guess what puzzled me is why `alluxio.user.hostname` can be deprecated by `alluxio.locality.node` as  `alluxio.locality.node` seems to apply to client, master and worker

Reply:
I think we tried to consolidate the concepts to only use `alluxio.locality.node` (this is a superset of the functionality provided by `alluxio.user.hostname`). On the client, the functionality is equivalent, and `alluxio.user.hostname` should not be set on the servers. However the property key itself is not very descriptive of what it does so I think the alias is fine.